base/uk.ac.stfc.isis.ibex.ui.configserver/src/uk/ac/stfc/isis/ibex/ui/configserver/commands/ConfigHandler.java

```diff
@@ -91,4 +96,51 @@ protected IWorkbenchWindow activeWindow() {
      */
 	public abstract void canWriteChanged(boolean canWrite);
 	
+    /**
+     * Implements the required eclipse interface.
+     * 
+     * Declared as final here, implement executeAndCatchErrors instead so that
+     * we can "hook into" any errors that occur in any of these dialogs and do
+     * something sensible with them.
+     * 
+     * @param event
+     *            the execution event
+     * @return null (ignored by eclipse framework)
+     */
+    @Override
+    public final Object execute(ExecutionEvent event) {
+        try {
+            safeExecute(event);
+        } catch (RuntimeException e) {
+            onError(e);
+        }
+        return null;
+    }
+
+    /**
+     * Executes an eclipse event. Any runtime errors thrown by this method will
+     * be caught by the execute() method and cause the onError() handler to be
+     * called.
+     * 
+     * @param event
+     *            the eclipse event
+     */
+    public abstract void safeExecute(ExecutionEvent event);
```

Comment:
This is really well done, considering the general case instead of just the specific case.